src/foam/nanos/theme/SubdomainAwareDAO.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.theme',
+  name: 'SubdomainAwareDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'DAO that adds value to subdomains property',
+
+  javaImports: [
+    'javax.servlet.http.HttpServletRequest'
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+        if ( obj instanceof SubdomainAware ) {
+          if ( ((SubdomainAware) obj).getSubdomain().isEmpty() ) {
+            var host = x.get(HttpServletRequest.class).getServerName();
```

Comment:
Do we want the raw host or the theme domain?  In development this will be localhost.  Also, should the subdomain be set to the subdomain, for example:  nbp.nanopay.net, the subdomain is npb.  
intuit.nanopay.net is intuit
treviso.nanopay.net is treviso  for example. 
In many cases this matches a spid.